src/UI/Components/FilterForm/index.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from "react";
+import { ToolbarGroup, ToolbarItem } from "@patternfly/react-core";
+import { Resource } from "@/Core";
+import { words } from "@/UI/words";
+import { FreeTextFilter } from "./FreeTextFilter";
+
+type Properties = "type" | "agent" | "value";
+interface Props {
+  filter: Resource.Filter;
+  setFilter: (filter: Resource.Filter) => void;
+}
+export const FilterForm: React.FC<Props> = ({ filter, setFilter }) => {
+  const [typeTextInput, setTypeTextInput] = useState("");
+  const [agentTextInput, setAgentTextInput] = useState("");
+  const [valueTextInput, setValueTextInput] = useState("");
+  const updateType = (types: string[], property: Properties) => {
+    return setFilter({
+      ...filter,
+      [property]: types.length > 0 ? types : undefined,
+    });
+  };
+  const removeChip = (id, prop: Properties) => {
+    //assingment is neccesary to avoid ts error due to it's limitation to recognize square brackets property
+    const p = filter[prop];
+    updateType(p ? p.filter((value) => value !== id) : [], prop);
+  };
+  const onTextInput = (event) => {
+    event.preventDefault();
+    let newFilter = { ...filter };
+    if (
+      typeTextInput.length <= 0 &&
```

Comment:
isnt `typeTextInput.length <= 0` the same as `!typeTextInput.length` ?

Reply:
Yeah, it actually works the same, I'll change that